### PR TITLE
PRIMES Assignment - Burgers RAR explanation

### PR DESCRIPTION
**Pull request by bfan05**

### CODE REVIEW

**Comment by lululxvi** [comment]
- Rename the file to burgers.rar.rst
- Also modify forward.rst

File: docs/demos/burgers_rar.rst [hunk 1]
```diff
@@ -0,0 +1,162 @@
+Burgers equation
```
Comment (lululxvi) [inline docs/demos/burgers_rar.rst:1]: Burgers equation with RAR

File: docs/demos/forward.rst [hunk 2]
```diff
@@ -41,7 +41,7 @@ Time-dependent PDEs
    diffusion.1d.resample
 
 - `Heat equation <https://github.com/lululxvi/deepxde/blob/master/examples/heat_conduction_1d_uniform_bar.ipynb>`_
-- `Burgers equation with residual-based adaptive refinement (RAR) <https://github.com/lululxvi/deepxde/blob/master/examples/Burgers_RAR.py>`_
+- `Burgers equation with residual-based adaptive refinement (RAR) <https://github.com/bfan05/deepxde/blob/burgers_rar_add_doc/docs/demos/burgers.rar.rst>`_
```
Comment (lululxvi) [inline docs/demos/forward.rst:44]: burgers.rar

File: docs/demos/burgers.rar.rst [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+    bc = dde.DirichletBC(geomtime, lambda x: 0, lambda _, on_boundary: on_boundary)
+    ic = dde.IC(geomtime, lambda x: -np.sin(np.pi * x[:, 0:1]), lambda _, on_initial: on_initial)
+    
+Now, we have specified the geometry, PDE residual, and boundary/initial condition. We then define the ``TimePDE`` problem as
+
+.. code-block:: python
+
+    data = dde.data.TimePDE(geomtime, pde, [bc, ic], 
+                            num_domain=2540, num_boundary=80, num_initial=160)
```
Comment (lululxvi) [inline docs/demos/burgers.rar.rst:61]: 2540?

File: docs/demos/burgers.rar.rst [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+Now, we have the PDE problem and the network. We build a ``Model`` and choose the optimizer and learning rate:
+
+.. code-block:: python
+
+    model = dde.Model(data, net)
+    model.compile("adam", lr=1e-3)
+    
+   
+We then train the model for 15000 iterations:
```
Comment (lululxvi) [inline docs/demos/burgers.rar.rst:79]: 15000?

File: docs/demos/burgers.rar.rst [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    losshistory, train_state = model.train(epochs=10000)
+    
+After we train the network using Adam, we continue to train the network using L-BFGS to achieve a smaller loss:
+
+.. code-block:: python
+
+    model.compile("L-BFGS-B")
+    losshistory, train_state = model.train()      
+
+Residual-based Adaptive Refinement (RAR) Implementation
```
Comment (lululxvi) [inline docs/demos/burgers.rar.rst:92]: Delete lines 92-93

File: docs/demos/burgers.rar.rst [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+-------------------------------------------------------
+
+Now that we have trained the model, we implement the residual-based adaptive refinement (RAR) method. This method is particularly effective with the Burgers equation because of the sharp discontinuity; intuitively, we should put more points near the sharp front to capture the discontinuity well. First, we randomly generate 100000 points from our domain to calculate the PDE residual. We also initialize the error to 1.
```
Comment (lululxvi) [inline docs/demos/burgers.rar.rst:92]: Because we only use 2500 residual points for training, the accuracy is low. Next, we improve the accuracy by the residual-based adaptive refinement (RAR) method. Because the Burgers equation has a sharp front, intuitively, we should put more points near the sharp front. First, we randomly generate 100000 points from our domain to calculate the PDE residual.

File: docs/demos/burgers.rar.rst [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+    while err > 0.005:
+        f = model.predict(X, operator=pde)
+        err_eq = np.absolute(f)
+        err = np.mean(err_eq)
+        print("Mean residual: %.3e" % (err))
+
+Next, we find the points where the residual is greatest and add these new points for training PDE loss. Furthermore, we define a callback function to check whether the network converges. 
+If there is significant improvement in the model's accuracy, as judged by the callback function, we continue to train the model.
+
+.. code-block:: python
```
Comment (lululxvi) [inline docs/demos/burgers.rar.rst:112]: Delete this block of code.

File: docs/demos/burgers.rar.rst [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+Because we only use 2500 residual points for training, the accuracy is low. Next, we improve the accuracy by the residual-based adaptive refinement (RAR) method. Because the Burgers equation has a sharp front, intuitively, we should put more points near the sharp front. First, we randomly generate 100000 points from our domain to calculate the PDE residual.
+
+.. code-block:: python
+
+    X = geomtime.random_points(100000)
+    err = 1
+
+We will repeatedly add points while the mean residual is greater than 0.005. Each iteration, we use our model to generate predictions for inputs in ``X`` and compute the absolute values of the errors. We then print the mean residual.
+
+.. code-block:: python
```
Comment (lululxvi) [inline docs/demos/burgers.rar.rst:101]: Delete this block of code. Merge these 3 paragraphs.
